core/src/data_sources/data_source.rs

```diff
@@ -404,21 +431,36 @@ impl DataSource {
                 &remove_tags,
             )
             .await?;
+        self.update_document_payload(qdrant_client, document_id, "tags", new_tags.clone())
+            .await?;
+        Ok(new_tags)
+    }
+
+    async fn update_document_payload(
+        &self,
+        qdrant_client: Arc<QdrantClient>,
+        document_id: String,
+        field_name: &str,
+        field_value: impl Into<Value>,
+    ) -> Result<()> {
         let mut payload = Payload::new();
-        payload.insert("tags", new_tags.clone());
+        payload.insert(field_name, field_value.into());
+
         let field_condition = qdrant::FieldCondition {
             key: "document_id".to_string(),
             r#match: Some(qdrant::Match {
                 match_value: Some(qdrant::r#match::MatchValue::Text(document_id)),
             }),
             ..Default::default()
         };
+
         let points_selector = PointsSelector {
             points_selector_one_of: Some(PointsSelectorOneOf::Filter(Filter {
                 must: vec![field_condition.into()],
                 ..Default::default()
             })),
         };
+
         qdrant_client
```

Comment:
Just to make sure we're sure we're not deleting the other pre-existing fiels here right?

(Also if you had called it `upsert` it would have hinted that we do :))

I would call it `update_document_payload_key` or `update_document_payload_field` to make it super clear?

Reply:
Yes, sure. It was a big topic during the timettamp migration, I had checked it at the time

Reply:
Roger that.